lib/jekyll/configuration.rb

```diff
@@ -377,15 +292,12 @@ def check_maruku(config)
     # Raises a Jekyll::Errors::InvalidConfigurationError if the config `plugins`
     # is a string
     def check_plugins(config)
-      if config.key?("plugins") && config["plugins"].is_a?(String)
-        Jekyll.logger.error "Configuration Error:", "You specified the" \
-          " `plugins` config in your configuration file as a string, please" \
-          " use an array instead. If you wanted to set the directory of your" \
-          " plugins, use the config key `plugins_dir` instead."
-        raise Jekyll::Errors::InvalidConfigurationError,
-              "'plugins' should not be a string, but was: " \
-              "#{config["plugins"].inspect}. Use 'plugins_dir' instead."
-      end
+      return unless config.key?("plugins")
+      return unless config["plugins"].is_a?(String)
+
+      raise Jekyll::Errors::InvalidConfigurationError,
+            "'plugins' should be set as an array, but was: #{config["plugins"].inspect}. " \
+            "Use 'plugins_dir' instead to set the directory of your plugins."
```

Comment:
Maybe explain that one key if for gems, and the other is for Ruby files here?

```
Error: Gems plugins should be listed in the `plugins` array (#{config["plugins"].inspect} detected).
Use `plugins_dir` instead to set the directory for Ruby plugins. (`_plugins` by default)
```

Reply:
Yeah.. that's a good idea :+1: